RecoMET/METAlgorithms/src/GlobalHaloAlgo.cc

```diff
@@ -231,5 +233,328 @@ reco::GlobalHaloData GlobalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeomet
     }
   
   TheGlobalHaloData.SetMETCorrections(dMEx, dMEy);
+
+  
+
+  std::vector<HaloClusterCandidateEB> hccandEB= TheEcalHaloData.getHaloClusterCandidatesEB();
+  std::vector<HaloClusterCandidateEE> hccandEE= TheEcalHaloData.getHaloClusterCandidatesEE();
+  std::vector<HaloClusterCandidateHB> hccandHB= TheHcalHaloData.getHaloClusterCandidatesHB();
+  std::vector<HaloClusterCandidateHE> hccandHE= TheHcalHaloData.getHaloClusterCandidatesHE();
+
+  //CSC-calo matching
+  bool ECALBmatched(false), ECALEmatched(false),HCALBmatched(false),HCALEmatched(false);
+
+  if (TheCSCSegments.isValid()) {
+    for(CSCSegmentCollection::const_iterator iSegment = TheCSCSegments->begin();
+	iSegment != TheCSCSegments->end();
+	iSegment++) {
+      
+      CSCDetId iCscDetID = iSegment->cscDetId();
+      bool Segment1IsGood=true;
+      
+      
+      //avoid segments from collision muons
+      if( TheMuons.isValid() )
+	{
+	  for(reco::MuonCollection::const_iterator mu = TheMuons->begin(); mu!= TheMuons->end() && (Segment1IsGood)   ; mu++ )
+	    {
+
+	      if( !mu->isTrackerMuon() && !mu->isGlobalMuon() && mu->isStandAloneMuon() ) continue;
+	      if( !mu->isGlobalMuon() &&  mu->isTrackerMuon() &&  mu->pt()<3) continue; 
+	      const std::vector<MuonChamberMatch> chambers = mu->matches();
+	      for(std::vector<MuonChamberMatch>::const_iterator kChamber = chambers.begin();
+		  kChamber != chambers.end(); kChamber ++ )
+		{
+		  if( kChamber->detector() != MuonSubdetId::CSC ) continue;
+		  for( std::vector<reco::MuonSegmentMatch>::const_iterator kSegment = kChamber->segmentMatches.begin();
+		       kSegment != kChamber->segmentMatches.end(); kSegment++ )
+		    {
+		      edm::Ref<CSCSegmentCollection> cscSegRef = kSegment->cscSegmentRef;
+		      CSCDetId kCscDetID = cscSegRef->cscDetId();
+		      
+		      if( kCscDetID == iCscDetID ) 
```

Comment:
Not sure, I will need to check that in the future (this was like this in the code before I started to maintain it).